src/ui/public/index_patterns/_index_pattern.js

```diff
@@ -110,7 +113,11 @@ export default function IndexPatternFactory(Private, Notifier, config, kbnIndex,
 
     if (!indexPattern.fields || !containsFieldCapabilities(indexPattern.fields)) {
       promise = indexPattern.refreshFields();
+    } else if (indexPattern.metaFieldsVersion <= 0) {
```

Comment:
Changing this to `<=` prevents the check from catching undefined metaFieldsVersion. Reverting it to a falsy `!indexPattern.metaFieldsVersion` check should fix it.